src/api_server/src/lib.rs

```diff
@@ -172,13 +185,45 @@ impl ApiServer {
         }
     }
 
-    fn serve_vmm_action_request(&self, vmm_action: Box<VmmAction>) -> Response {
+    fn serve_vmm_action_request(
+        &self,
+        vmm_action: Box<VmmAction>,
+        action_start_us: u64,
+    ) -> Response {
+        let metric_with_action = match *vmm_action {
+            #[cfg(target_arch = "x86_64")]
+            VmmAction::CreateSnapshot(ref params) => match params.snapshot_type {
+                SnapshotType::Full => Some((
+                    &METRICS.latencies_us.full_create_snapshot,
+                    "creating the full snapshot",
```

Comment:
nit:
Since these are debug messages (developer-oriented), they can be shorter and not necessarily prettified:
```suggestion
                    "create full snapshot",
```

`load snapshot`
`pause vm`
`resume vm`
etc

Reply:
Done.